code/game/machinery/iv_drip.dm

```diff
@@ -112,12 +112,14 @@
 	if(src.attached && src.beaker)
 		// Give blood
 		if(mode)
-			if(src.beaker.volume > 0)
-				var/transfer_amount = REAGENTS_METABOLISM
+			if(beaker.volume > 0)
 				if(beaker.reagents.reagent_list.len == 1 && beaker.reagents.has_reagent(BLOOD))
 					// speed up transfer if the container has ONLY blood
-					transfer_amount = 4
-				src.beaker.reagents.trans_to(src.attached, transfer_amount)
+					beaker.reagents.trans_to(attached, 4)
+				else
+					// otherwise: transfer a little bit of all reagents to the patient. the reason why we don't transfer a set amount is because 0.2u of 10 different reagents is 0.02u of each, which is entirely too little.
+					for(var/datum/reagent/reagent in beaker.reagents.reagent_list)
```

Comment:
istypeless loop this meme, if there's anything other than reagents in here you've got bigger problems